api/django-common/django_common/models.py

```diff
@@ -22,7 +22,7 @@ class OwnedModel(models.Model):
     A model that has a relationship to the owner in the user model.
     """
 
-    owner = models.ForeignKey(get_user_model(), on_delete=models.RESTRICT)
+    owner = models.ForeignKey(get_user_model(), on_delete=models.RESTRICT, null=True, blank=True)
```

Comment:
Smell: Inherit from a class that has only optional members.

Reply:
This is basically tied to the above question about having an owner. The problem is that if the model expects a django user there is no possibility of having an anonymous upload.

Reply:
Please indicate in a comment that `null` stands for the anonymous user and it will be OK.

Reply:
Even better than a comment is to force it. Here is some pseudocode:

```
def __setattr__(self, name, value):
    if name == "owner":
        if isinstance(value, User) or isinstance(value, AnonymousUser):
            return super().__setattr__(name, value.id)
        raise ValueError
    else:
        return super().__setattr__(name, value)
```

In my opinion, a good API is one that forces you to use it as intended. In order to set the owner member, a valid user object must be provided.

Reply:
So I attempted to limit the users to AnonymousUser and User. However, it seems that the OwnedModel already has a ForeignKey constraint that limits the type of user to Users. So the only option is to use None instead. Therefore I left the comment and will also provide this in the documentation.